src/test/java/jenkins/plugins/git/GitSCMBuilderTest.java

```diff
@@ -0,0 +1,598 @@
+                instanceOf(AuthorInChangelog.class),
+                instanceOf(GitSCMSourceDefaults.class),
+                instanceOf(CleanCheckout.class)
+        ));
+
+        instance.withExtension(new LocalBranch("**"));
+        assertThat(instance.extensions(), containsInAnyOrder(
+                instanceOf(AuthorInChangelog.class),
+                instanceOf(CleanCheckout.class),
+                allOf(instanceOf(LocalBranch.class), hasProperty("localBranch", is("**")))
+        ));
+        scm = instance.build();
+        assertThat(scm.getBrowser(), is(nullValue()));
+        assertThat(scm.getUserRemoteConfigs(), contains(allOf(
+                instanceOf(UserRemoteConfig.class),
+                hasProperty("url", is("http://git.test/repo.git")),
+                hasProperty("name", is("origin")),
+                hasProperty("refspec", is("+refs/heads/*:refs/remotes/origin/*")),
+                hasProperty("credentialsId", is(nullValue())))
+        ));
+        assertThat(scm.getGitTool(), is(nullValue()));
+        assertThat(scm.getExtensions(), containsInAnyOrder(
+                instanceOf(AuthorInChangelog.class),
+                instanceOf(GitSCMSourceDefaults.class),
+                instanceOf(CleanCheckout.class),
+                allOf(instanceOf(LocalBranch.class), hasProperty("localBranch", is("**")))
+        ));
+
+        // repeated calls re-define up existing extensions
+        instance.withExtension(new LocalBranch("master"));
+        assertThat(instance.extensions(), containsInAnyOrder(
+                instanceOf(AuthorInChangelog.class),
+                instanceOf(CleanCheckout.class),
+                allOf(instanceOf(LocalBranch.class), hasProperty("localBranch", is("master")))
+        ));
+        scm = instance.build();
+        assertThat(scm.getBrowser(), is(nullValue()));
+        assertThat(scm.getUserRemoteConfigs(), contains(allOf(
+                instanceOf(UserRemoteConfig.class),
+                hasProperty("url", is("http://git.test/repo.git")),
+                hasProperty("name", is("origin")),
+                hasProperty("refspec", is("+refs/heads/*:refs/remotes/origin/*")),
+                hasProperty("credentialsId", is(nullValue())))
+        ));
+        assertThat(scm.getGitTool(), is(nullValue()));
+        assertThat(scm.getExtensions(), containsInAnyOrder(
+                instanceOf(AuthorInChangelog.class),
+                instanceOf(GitSCMSourceDefaults.class),
+                instanceOf(CleanCheckout.class),
+                allOf(instanceOf(LocalBranch.class), hasProperty("localBranch", is("master")))
+        ));
+    }
+
+    @Test
+    public void withGitTool() throws Exception {
+        instance.withGitTool("git");
+        assertThat(instance.gitTool(), is("git"));
+        GitSCM scm = instance.build();
+        assertThat(scm.getBrowser(), is(nullValue()));
+        assertThat(scm.getUserRemoteConfigs(), contains(allOf(
```

Comment:
Seems very excessive to assert on these all the time, since if there is a problem with it it will get reported on "unrelated" tests.